src/client/app/components/admin/users/CreateUserModalComponent.tsx

```diff
@@ -0,0 +1,245 @@
+			...prevDetails,
+			passwordMatch: (userDetails.password === userDetails.confirmPassword),
+			passwordLength: userDetails.password.length > 7
+		}));
+	}, [userDetails.password, userDetails.confirmPassword]);
+
+	// check if form is valid
+	const isFormValid = () => {
+		return !!userDetails.username &&
+			!!userDetails.password &&
+			(userDetails.confirmPassword === userDetails.password) &&
+			!!(userDetails.role !== 'invalid');
+	};
+
+	// Handlers for each type of input change
+	const handleStringChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setUserDetails(prevDetails => ({
+			...prevDetails,
+			[e.target.name]: e.target.value
+		}));
+	};
+
+	const handleRoleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setUserDetails(prevDetails => ({
+			...prevDetails,
+			role: e.target.value as UserRole
+		}));
+	};
+
+	// Methods to reset form fields
+	const resetForm = () => {
+		setUserDetails(userDefaults);
+	};
+
+	const resetPasswordFields = () => {
+		setUserDetails({
+			...userDetails, password: '',
+			confirmPassword: ''
+		});
+	};
+	// End of reset form methods
+
+	// Modal show/close
+	const [showModal, setShowModal] = useState(false);
+
+	const handleShowModal = () => setShowModal(true);
+
+	const handleCloseModal = () => {
+		resetPasswordFields();
+		setShowModal(false);
+	};
+	// End Modal show/close
+
+	const handleSubmit = async () => {
+		const newUser: User = { username: userDetails.username, role: userDetails.role, password: userDetails.password, note: userDetails.note };
+		createUser(newUser)
+			.unwrap()
+			.then(() => {
+				showSuccessNotification(translate('users.successfully.create.user') + userDetails.username);
+				resetForm();
+				handleCloseModal();
+			})
+			.catch(error => {
+				showErrorNotification(translate('users.failed.to.create.user') + userDetails.username + ' ' + error.data.message);
+				resetPasswordFields();
+			});
+	};
+
+	// allow enter key to submit form
+	const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+		if (event.key === 'Enter') {
+			event.preventDefault();
+			if (isFormValid()) {
+				handleSubmit();
+			} else {
+				console.log('Form is not valid');
+			}
+		}
+	};
+
+	const tooltipStyle = {
+		...tooltipBaseStyle,
+		// Only an admin can create a meter.
+		tooltipUsersView: 'help.admin.users'
+	};
+
+	return (
+		<>
+			<Button color="secondary" onClick={handleShowModal}>
+				{translate('create.user')}
+			</Button>
+			<Modal isOpen={showModal} toggle={handleCloseModal} size="lg">
+				<ModalHeader>
+					{translate('create.user')}
+					<TooltipHelpComponent page='users-create' />
+					<div style={tooltipStyle}>
+						<TooltipMarkerComponent page='users-create'
+							helpTextId={tooltipStyle.tooltipUsersView}
+						/>
+					</div>
+				</ModalHeader>
+				<ModalBody>
+					<Container onKeyDown={handleKeyDown}>
+						<Row xs='1' lg='2'>
+							<Col>
+								<FormGroup>
+									<Label for="username">
```

Comment:
I'm not sure why it passed the checks but all of these should be single quotes.

Reply:
Will do.

Reply:
Completed